src/engine/GroupBy.cpp

```diff
@@ -764,16 +786,23 @@ void GroupBy::computeResult(ResultTable* result) {
 
   int inWidth = subresult->_data.cols();
   int outWidth = result->_data.cols();
-  CALL_FIXED_SIZE_2(inWidth, outWidth, doGroupBy, subresult->_data,
-                    inputResultTypes, groupByCols, aggregates, &result->_data,
-                    subresult.get(), result, getIndex());
-
-  // Free the user data used by GROUP_CONCAT aggregates.
-  for (Aggregate& a : aggregates) {
-    if (a._type == ParsedQuery::AggregateType::GROUP_CONCAT) {
-      delete static_cast<std::string*>(a._userData);
+
+  auto cleanup = [&]() {
```

Comment:
In general you are right. In this case I would argue that the default capture by reference is not problematic, because we do not pass the lambda outside, and only use it as a "copied code block", but with NEVER using default captures you agree with one of my favourite C++ authors, so I am totally convinced.